stamp/preprocessing/wsi_norm.py

```diff
@@ -235,6 +256,9 @@ def preprocess(output_dir: Path, wsi_dir: Path, model_path: Path, cache_dir: Pat
                         print("Deleting slide from local folder...")
                         if os.path.exists(slide_url):
                             os.remove(slide_url)
+                    
+                    if preload_wsi:
+                        shutil.rmtree(slide_tmp_dir)
```

Comment:
The temporary directory and the files it contains are only getting removed when none of the previous exceptions with a continue at the end are triggered. So maybe the usage of `tempfile.TemporaryDirectory()` as a context manager after the `with lock_file(slide_url), ...:` could be an option to ensure the automatic deletion after the preprocessing of a file has finished.